include/libcyphal/types.hpp

```diff
@@ -50,12 +51,54 @@ using TimePoint = MonotonicClock::time_point;
 using Duration  = MonotonicClock::duration;
 
 template <typename T>
-using UniquePtr = cetl::pmr::Factory::unique_ptr_t<cetl::pmr::polymorphic_allocator<T>>;
+using UniquePtr = cetl::pmr::InterfacePtr<T>;
 
 // TODO: Maybe introduce `cetl::expected` at CETL repo.
 template <typename Success, typename Failure>
 using Expected = cetl::variant<Success, Failure>;
 
+/// A generalized implementation of https://www.fluentcpp.com/2021/01/29/inheritance-without-pointers/
+/// that works with any `cetl::unbounded_variant`.
+///
+/// The instance is always initialized with a valid value, but it may turn valueless if the value is moved.
+/// The AnyUbVar type can be a `cetl::unbounded_variant`.
+///
+template <typename Interface, typename AnyUbVar>
```

Comment:
Shouldn't we call this any just `Any`? Looking at the usages I think any any would fit, like `std::any`.